### PR TITLE
Test: e2e Cypress migration from ODS-2206

### PR DESCRIPTION
**Pull request by FedeAlonso**
https://issues.redhat.com/browse/RHOAIENG-16476
## Description
Cypress Smoke test from ODS-CI "ODS-2206"
Title: Verify User Can Create, Run and Delete A DS Pipeline From DS Project Details Page Using Custom Pip Mirror 
This test Run an Iris DSP, and then deletes the DSP version and the DSP. All just by using the UI.
It has also the particularity that the PIP server (or mirror) is configured by env variables, so it can run in a disconnected environment.

## How Has This Been Tested?
Locally against a live cluster.
Command: `$ npx cypress run --spec "**/createRunDeletePipelineCustomPipMirror.cy.ts" --browser chrome`
image

## Test Impact
Addition of a new e2e Cypress test

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

After the PR is posted & before it merges:
- [x] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Review by manosnoam — COMMENTED** [review]
/lgtm

**Comment by FedeAlonso** [comment]
Tested with thhe latest changes:
image

**Comment by FedeAlonso** [comment]
Tested with latest changes: 
image

File: frontend/src/__tests__/cypress/cypress/pages/components/table.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export class TableRow extends Contextual<HTMLTableRowElement> {
   }
 
   findKebabAction(name: string): Cypress.Chainable<JQuery<HTMLElement>> {
-    return this.find().findKebabAction(name);
+    return this.find().findKebabAction(name).should('exist').and('be.visible');
```
Comment (antowaddle) [inline frontend/src/__tests__/cypress/cypress/pages/components/table.ts:20]: Can you check please if this is breaking the Mock Tests? 
https://github.com/opendatahub-io/odh-dashboard/actions/runs/12427145849/job/34696500427?pr=3580

File: frontend/src/__tests__/cypress/cypress/tests/e2e/dataSciencePipelines/createRunDeletePipelineCustomPipMirror.cy.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+const testRunName = 'test-pipelines-run';
+
+describe('An admin user can import and run a pipeline', { testIsolation: false }, () => {
+  before(() => {
+    provisionProjectForPipelines(projectName, dspaSecretName);
+    const pipConfig = Cypress.env('PIP_CONFIG');
+    //Create Pipelines ConfigMap With Custom Pip Index Url And Trusted Host
+    createOpenShiftConfigMap('ds-pipeline-custom-env-vars', projectName, {
+      // The following lines should be snake case
+      /* eslint-disable-next-line camelcase */
```
Comment (antowaddle) [inline frontend/src/__tests__/cypress/cypress/tests/e2e/dataSciencePipelines/createRunDeletePipelineCustomPipMirror.cy.ts:32]: Should these eslint lines be in here?

File: frontend/src/__tests__/cypress/test-variables.yml.example [hunk 3]
```diff
@@ -17,4 +17,7 @@ S3:
   BUCKET_2:
     NAME: pipeline-bucket-name
     REGION: pipeline-bucket-region
-    ENDPOINT: https://pipeline-bucket-endpoint.com/
\ No newline at end of file
+    ENDPOINT: https://pipeline-bucket-endpoint.com/
```
Comment (antowaddle) [inline frontend/src/__tests__/cypress/test-variables.yml.example:20]: Is another PR required to apply these variables to Jenkins prior to merging?

**Review by antowaddle — COMMENTED** [review]
@FedeAlonso Can you also please check this test against the latest ODH Nightly build? The test also currently takes 4 minutes to execute, is there any way that time can be improved? Or is there any type of splitting that can be done?

File: frontend/src/__tests__/cypress/cypress/tests/e2e/dataSciencePipelines/createRunDeletePipelineCustomPipMirror.cy.ts (hunk 2 above)
Comment (FedeAlonso) [reply frontend/src/__tests__/cypress/cypress/tests/e2e/dataSciencePipelines/createRunDeletePipelineCustomPipMirror.cy.ts:32]: Yes, those are in order to exclude one rule of the linter, as the variable name should be snake case

**Comment by FedeAlonso** [comment]
> @FedeAlonso Can you also please check this test against the latest ODH Nightly build? The test also currently takes 4 minutes to execute, is there any way that time can be improved? Or is there any type of splitting that can be done?

This test takes 4 minutes because it's a quite complete one. Also, running a pipeline always takes time, and in this case we need to have a more complex one, as it's used also for disconnected testing.

**Comment by FedeAlonso** [comment]
image
Test results

File: frontend/src/__tests__/cypress/cypress/pages/components/table.ts (hunk 1 above)
Comment (FedeAlonso) [reply frontend/src/__tests__/cypress/cypress/pages/components/table.ts:20]: Fixed

**Comment by FedeAlonso** [comment]
Latest changes
image

**Review by manosnoam — APPROVED** [review]

**Pull request merged**